data/account/participation.go

```diff
@@ -58,27 +57,36 @@ type Participation struct {
 	KeyDilution uint64
 }
 
+// participationIDData is for msgpack encoding the participation data.
+type participationIDData struct {
+	_struct     struct{} `codec:""`
+	Parent      basics.Address
+	VRF         crypto.VRFSecrets
+	FirstValid  basics.Round
+	LastValid   basics.Round
+	KeyDilution uint64
```

Comment:
I'd suggest you'll try to keep the same standard codec definitions we have on this object as well.

Reply:
Pointers don't seem to be supported, which was causing the generated msgp tests to fail. Is there a standard convenient for dealing with that?

Reply:
let me take a look on that one locally and I'll let you know ?

Reply:
so, instead of having the `crypto.VRFSecrets` here, use just the `SK` field `crypto.VrfPrivkey`. The `PK` is derived from the `SK` anyway.

Reply:
Maybe we can chat next week. What I have here is technically working as designed, but I had to redefine the `Participation` type. `Participation` didn't work properly because the generated marshalling function didn't seem to work properly with pointer types.

Reply:
see - https://github.com/winder/go-algorand/pull/4

Reply:
Thanks, updated.